extensions/EXT/GLX_EXT_no_config_context.txt

```diff
@@ -0,0 +1,254 @@
+    compatible with (subject to the restrictions discussed in the section on
+    address space and the restrictions discussed under glXCreatePixmap). A
+    drawable and context are compatible if either of the following sets of
+    conditions are true:
+
+    * The context was created with no GLXFBConfig
+
+    * The drawable and context are on the same screen
+
+    or:
+
+    [...]"
+
+    Append a new paragraph to the end of section 2.1 "Rendering Contexts
+    and Drawing Surfaces":
+
+    "If GLX_NO_CONFIG_EXT was specified as the GLXFBConfig at creation, the
+    resulting context is said to be created <without reference to a config>.
+    In this case, the context must pass the required conformance tests for
+    that client API and must support being made current without a rendering
+    surface. Such support is guaranteed for OpenGL ES 2.0 implementations
+    supporting the GL_OES_surfaceless_context extension, OpenGL ES 3.0 and
+    later versions of OpenGL ES, and OpenGL 3.0 and later versions of OpenGL.
+    Support for other versions and other client APIs is implementation
+    dependent. Such contexts are compatible with any drawable on the same
+    screen, though the API may impose further restrictions. For example,
+    OpenGL 3.0 contexts can not render to color index surfaces, even if color
+    index GLXFBConfigs are available."
+
+    In section 3.3.7 "Rendering Contexts" in the discussion of glXQueryContext,
+    to the paragraph beginning "The values and types", append:
+
+    "If the context was created without reference to a config, the value
+    of GLX_RENDER_TYPE will be GLX_DONT_CARE, and the values of
+    GLX_FBCONFIG_ID and GLX_VISUAL_ID_EXT will be GLX_NO_CONFIG_EXT."
+
+Interactions with GLX_ARB_fbconfig_float
+
+    A configuration created without reference to a config may be used to
+    render to surfaces of GLX_RGBA_FLOAT_TYPE_ARB in addition to RGBA
+    and color-index surfaces.
+
+Interactions with GLX_EXT_fbconfig_packed_float
+
+    A configuration created without reference to a config may be used to
+    render to surfaces of GLX_RGBA_UNSIGNED_FLOAT_TYPE_EXT in addition to
+    RGBA and color-index surfaces.
+
+Errors
+
+    No new error types are added.
+
+GLX Protocol
+
+    No new requests are added.
+
+    For GLXCreateContextAttribsARB, normally the client library derives the
+    <screen> field in the request from the specified fbconfig. When
+    GLX_NO_CONFIG_EXT is given as the fbconfig, the GLX_SCREEN attribute
+    and the <screen> field must match, or BadValue is generated. This implies
+    that the client library is responsible for pulling the screen number out
+    of the attribute list, since glXCreateContextAttribsARB does not have an
+    explicit screen parameter.
+
+Issues
+
+ 1) Should non-conformant no-config contexts be allowed to be created?
+
+    RESOLVED: No. We are not encouraging non-conformant contexts.
```

Comment:
This was just copypasta from the EGL spec, I have no problem deleting it. Other GLXes have had non-conformant caveats on the ARGB visual, which is in some sense mandatory for modern X, so this would more be saying "we don't have a good definition for conformance here" than encouraging bad GL, I think.